Flaky TestDeltaLakeDeleteCompatibility.testDeletionVectors on 13.3 LTS

```
io.trino.tests.product.deltalake.TestDeltaLakeDeleteCompatibility.testDeletionVectors (Groups: profile_specific_tests, delta-lake-exclude-91, delta-lake-databricks, delta-lake-exclude-104, delta-lake-exclude-113, delta-lake-oss) took 18.2 seconds
tests               | 2023-09-02 17:21:00 SEVERE: Failure cause:
tests               | java.lang.AssertionError: Expected row count to be <3>, but was <4>; rows=[[0, CREATE TABLE], [1, WRITE], [2, DELETE], [3, OPTIMIZE]]
tests               | 	at io.trino.tests.product.deltalake.TestDeltaLakeDeleteCompatibility.testDeletionVectors(TestDeltaLakeDeleteCompatibility.java:147)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
tests               | 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
tests               | 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
tests               | 	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
tests               | 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
tests               | 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
tests               | 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
tests               | 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
tests               | 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
tests               | 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
tests               | 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
tests               | 	at java.base/java.lang.Thread.run(Thread.java:833)
```

https://github.com/trinodb/trino/actions/runs/6057878380/job/16439570813

--- a/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDeleteCompatibility.java
+++ b/testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDeleteCompatibility.java
@@ -144,7 +144,8 @@ public class TestDeltaLakeDeleteCompatibility
             assertThat(onTrino().executeQuery("SHOW TABLES FROM delta.default"))
                     .contains(row(tableName));
             assertThat(onTrino().executeQuery("SELECT version, operation FROM delta.default.\"" + tableName + "$history\""))
-                    .containsOnly(row(0, "CREATE TABLE"), row(1, "WRITE"), row(2, "DELETE"));
+                    // Use 'contains' method because newer Databricks clusters execute OPTIMIZE statement in the background
+                    .contains(row(0, "CREATE TABLE"), row(1, "WRITE"), row(2, "DELETE"));
             assertThat(onTrino().executeQuery("SELECT comment FROM information_schema.columns WHERE table_schema = 'default' AND table_name = '" + tableName + "'"))
                     .hasNoRows();
             assertQueryFailure(() -> onTrino().executeQuery("SELECT * FROM delta.default." + tableName))

